src/utils.ts

```diff
@@ -2,4 +2,48 @@ const isValueEmpty = (value: string | null | undefined): boolean => {
   return value === null || value === undefined || value.trim().length == 0;
 };
 
-export { isValueEmpty };
+function deepEquals(x: any, y: any) {
```

Comment:
I'd really like a link to more information about the mechanisms here :) It makes sense, but if I needed to verify anything I'd have to start digging from zero.

Reply:
This is practically proved by tests and theoretically by stackoverflow :)

Reply:
https://stackoverflow.com/questions/1068834/object-comparison-in-javascript